jax/_src/xla_bridge.py

```diff
@@ -246,6 +247,64 @@ def make_cpu_client() -> xla_client.Client:
     "cpu", make_cpu_client, priority=0, fail_quietly=False
 )
 
+def _check_ptxas_compat() -> None:
+  import re
+  import subprocess
+  from glob import glob
+
+  cuda_path = None
+
+  # The pip-installed nvidia packages may have been installed in a separate
+  # location from JAX so we check for it specifically.
+  try:
+    import nvidia # type: ignore
+    nvidia_path = Path(nvidia.__file__).parent
+    cuda_path = nvidia_path.joinpath(Path("cuda_nvcc/bin"))
+  except ModuleNotFoundError:
+    pass
+
+  searchable_paths = []
+  if cuda_path:
+    searchable_paths += [cuda_path]
+  from_PATH = os.environ["PATH"].split(os.pathsep)
+  searchable_paths += [Path(path) for path in from_PATH]
+  found_ptxas = None
+  for path in searchable_paths:
+      ptxas_relpath = glob("**/ptxas", root_dir=path)
+      if ptxas_relpath:
+        found_ptxas = path / Path(ptxas_relpath[0])
+        break
+
+  if found_ptxas is None:
+    raise RuntimeError(
+      "ptxas binary not found -- please either install `nvidia-cuda-nvcc-cu12` via "
```

Comment:
Note: you'd need `nvidia-cuda-nvcc-cu11` is this is a CUDA 11 JAX; you should add this case to the error message.

Same below.